bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java

```diff
@@ -168,6 +220,117 @@ private void processIncomingState(String messageJSON) {
         }
     }
 
+    private boolean hasCCT() {
+        switch (globeType) {
+            case "rgb_cct":
+            case "cct":
+            case "fut089":
+            case "fut091":
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    private boolean hasRGB() {
+        switch (globeType) {
+            case "rgb_cct":
+            case "rgb":
+            case "rgbw":
+            case "fut089":
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    /**
+     * Scales mireds to 0-100%
+     */
+    private static PercentType scaleMireds(int mireds) {
+        // range in mireds is 153-370
+        // 100 - (mireds - 153) / (370 - 153) * 100
+        if (mireds >= 370) {
+            return PercentType.HUNDRED;
+        } else if (mireds <= 153) {
+            return PercentType.ZERO;
+        }
+        return new PercentType(BIG_DECIMAL_100.subtract(new BigDecimal(mireds).subtract(BIG_DECIMAL_153)
+                .divide(BIG_DECIMAL_217, MathContext.DECIMAL128).multiply(BIG_DECIMAL_100)));
+    }
+
+    private static BigDecimal polynomial_fit(BigDecimal x, BigDecimal[] coefficients) {
+        var result = BigDecimal.ZERO;
+        var x_accumulator = BigDecimal.ONE;
+        // forms K[4]*x^0 + K[3]*x^1 + K[2]*x^2 + K[1]*x^3 + K[0]*x^4
+        // (or reverse the order of terms for the usual way of writing it in academic papers)
+        int i = 0;
+        while (true) {
```

Comment:
it saves an extra BigDecimal.multiply on x_accumulator, but that's not a big deal. I've re-written as a for loop since it is a lot easier to understand.